### PR TITLE
[sil] Rename ValueOwnershipKindClassifier.cpp -> ValueOwnership.cpp a…

### PR DESCRIPTION
**Pull request by gottesmm**
…nd remove ValueOwnershipKindClassifier.h

This matches the name of OperandOwnership.cpp which performs an anologous
function. I also was able to eliminate an unneeded header.  The key thing here
is that the only reason that we had ValueOwnershipKindClassifier.h was because
we needed some defs from it in SILValue.cpp for SILValue::getOwnershipKind()
... which is great except for the fact that ValueOwnershipKindClassifier is
basically the implementation of SILValue::getOwnershipKind().

So what this patch does is moves the implementation of
SILValue::getOwnershipKind() from SILValue.cpp -> ValueOwnership.cpp and then
puts the visitor into an anonymous namespace in that file.

I also added a little comment in SILValue.h that says that
SILValue::getOwnershipKind() is implemented in ValueOwnership.cpp, not
SILValue.cpp.

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Review by atrick — APPROVED** [review]
Great.
